src/neighbors.cc

```diff
@@ -468,6 +465,10 @@ peerAlive(CachePeer *p)
 
     p->stats.last_reply = squid_curtime;
     p->stats.probe_start = 0;
+
+    // TODO: Remove or explain how we could detect an alive peer without IP addresses
+    if (!p->n_addresses)
+        ipcache_nbgethostbyname(p->host, peerDNSConfigure, p);
```

Comment:
This `if` statement is old code from peerConnectSucceded(). AFAICT, it either never runs at all (and should be removed) or runs only in some very rare, special cases (which we should document), and this move does not change any of that. If others insist on preserving the location of this code, we would have to add a new dedicated global function or make peerDNSConfigure() public/global, just to call it from CachePeer::noteSuccess(). Otherwise, peerDNSConfigure() would bring too many dependencies into CachePeer.cc that this PR does not care about.

Reply:
IMO should be out of scope for this PR. The whole probing and neighbors.cc logic needs a refactor, this if-statement is no different from the many other things which should (re)move at that time.

Reply:
Note that this `if` in the old code and new one has the same effect (p->n_addresses>0 on other peerAlive() callers). This 'out of scope' change is simply intended to minimize more of scope changes...

Reply:
Sorry for not being clear. I was agreeing with the choice to copy the broken if-statement and add TODO for later fix.